src/plugins/visualizations/public/embeddable/initialize_edit_api.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  PublishingSubject,
+  apiHasAppContext,
+  apiPublishesTimeRange,
+} from '@kbn/presentation-publishing';
+import { TimeRange } from '@kbn/es-query';
+import type { Vis } from '../vis';
+import { urlFor } from '../utils/saved_visualize_utils';
+import { getCapabilities, getEmbeddable } from '../services';
+
+export function initializeEditApi({
+  customTimeRange$,
+  description$,
+  parentApi,
+  savedObjectId$,
+  searchSessionId$,
+  title$,
+  vis$,
+  uuid,
+}: {
+  customTimeRange$: PublishingSubject<TimeRange | undefined>;
+  description$: PublishingSubject<string | undefined>;
+  parentApi?: unknown;
+  savedObjectId$: PublishingSubject<string | undefined>;
+  searchSessionId$: PublishingSubject<string | undefined>;
+  title$: PublishingSubject<string | undefined>;
+  vis$: PublishingSubject<Vis>;
+  uuid: string;
+}) {
+  return !parentApi || !apiHasAppContext(parentApi)
+    ? {}
+    : {
+        getTypeDisplayName: () =>
+          i18n.translate('visualizations.displayName', {
+            defaultMessage: 'visualization',
+          }),
+        onEdit: async () => {
+          const stateTransferService = getEmbeddable().getStateTransfer();
+          const visId = savedObjectId$.getValue();
+          const editPath = visId ? urlFor(visId) : '#/edit_by_value';
+          const parentTimeRange = apiPublishesTimeRange(parentApi)
+            ? parentApi.timeRange$.getValue()
+            : {};
+          const customTimeRange = customTimeRange$.getValue();
+          const parentApiContext = parentApi.getAppContext();
```

Comment:
This is the fix for the visualize embeddable. Calling `getAppContext` in the `onEdit` callback